### PR TITLE
Initial lock support

### PR DESCRIPTION
**Pull request by nappa85**
As discussed in https://github.com/SeaQL/sea-orm/issues/111, this PR adds the basis for row locking support.

As always, I'm open for discussion and improvement

### CODE REVIEW

File: src/query/select.rs [hunk 1]
```diff
@@ -1345,3 +1366,10 @@ impl ConditionalStatement for SelectStatement {
         self
     }
 }
+
+/// List of lock types
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum LockType {
```
Comment (tyt2y3) [inline src/query/select.rs:1372]: Very tiny comment here. I usually put type definitions before implementations.

File: src/backend/query_builder.rs [hunk 2]
```diff
@@ -338,6 +343,24 @@ pub trait QueryBuilder: QuotedBuilder {
...
+    fn prepare_select_lock(
+        &self,
+        select_lock: &LockType,
+        sql: &mut SqlWriter,
+        _collector: &mut dyn FnMut(Value),
+    ) {
+        write!(
+            sql,
+            "{}",
+            match select_lock {
```
Comment (tyt2y3) [inline src/backend/query_builder.rs:356]: This trait impl things that are common across MySQL, Postgres and SQLite. Seemingly we cannot provide a default impl here, as things are different for Postgres and SQLite (which has no row based lock indeed).

Comment (nappa85) [reply src/backend/query_builder.rs:356]: To be honest, I didn't check for sqlite before, and in fact now I find out it doesn't support row locking.
Postgres supports the same lock commands as MySQL, the docs are a little confusionary, all the lock types described are used internally.

I think we can leave this impl here and overwrite it in sqlite part

Comment (nappa85) [reply src/backend/query_builder.rs:356]: How do you disable features in a single DB flavour?
ATM I've put a panic, but I don't feel good with it, at the same time writing code that does nothing IMHO isn't good too.

File: src/query/select.rs (hunk 1 above)
Comment (nappa85) [reply src/query/select.rs:1372]: I've copied the comment style for SelectDistinct

File: src/backend/query_builder.rs (hunk 2 above)
Comment (tyt2y3) [reply src/backend/query_builder.rs:356]: I think 'do nothing' is the usual behavior

Comment (nappa85) [reply src/backend/query_builder.rs:356]: Ok, I've removed the panic

File: src/query/select.rs (hunk 1 above)
Comment (nappa85) [reply src/query/select.rs:1372]: Oh, I've realized only now you maybe intended the comments on the lock methods...
I was waiting for your approval on the design before spending time on docs and examples

Comment (tyt2y3) [reply src/query/select.rs:1372]: Yes, the API looks good to me.

Comment (nappa85) [reply src/query/select.rs:1450]: Ok, I've added some docs and examples

**Pull request merged**